tests/test_cos_integration.py

```diff
@@ -38,86 +47,57 @@ async def test_build_deploy_knative_charms(ops_test: OpsTest):
     )
 
     await ops_test.model.deploy(
-        knative_charms["knative-serving"],
-        application_name="knative-serving",
-        config={"namespace": "knative-serving", "istio.gateway.namespace": ops_test.model_name},
+        # knative_charms["knative-serving"],
+        # application_name="knative-serving",
+        "knative-serving",
+        channel="latest/edge",
+        config={
+            "namespace": f"{ops_test.model_name}-serving",
+            "istio.gateway.namespace": ops_test.model_name,
+        },
         trust=True,
     )
 
     await ops_test.model.deploy(
-        knative_charms["knative-eventing"],
-        application_name="knative-eventing",
-        config={"namespace": "knative-eventing"},
+        # knative_charms["knative-eventing"],
+        # application_name="knative-eventing",
+        "knative-eventing",
+        channel="latest/edge",
+        config={"namespace": f"{ops_test.model_name}-eventing"},
         trust=True,
     )
 
+    # Wait here to avoid a race condition between the rest of the tests and knative
+    # eventing/serving coming up. This race condition is because of:
+    # https://github.com/canonical/knative-operators/issues/50
```

Comment:
No, but I see that the issue is still there, so I kept it here. However, instead of waiting I used `idle_period`, which mean that charms needs to be in idle for at least 60s to not failed this function.